core/include/detray/core/detail/single_store copy.hpp

```diff
@@ -0,0 +1,151 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s)
+#include "detray/core/detail/container_views.hpp"
+#include "detray/core/detail/data_context.hpp"
+#include "detray/definitions/indexing.hpp"
+#include "detray/definitions/qualifiers.hpp"
+
+// Vecmem include(s)
+#include <vecmem/containers/data/jagged_vector_buffer.hpp>
+#include <vecmem/containers/data/jagged_vector_view.hpp>
+#include <vecmem/containers/data/vector_buffer.hpp>
+#include <vecmem/containers/data/vector_view.hpp>
+#include <vecmem/containers/device_vector.hpp>
+#include <vecmem/containers/jagged_device_vector.hpp>
+#include <vecmem/containers/jagged_vector.hpp>
+#include <vecmem/containers/vector.hpp>
+
+#include <vecmem/memory/memory_resource.hpp>
+#include <vecmem/utils/copy.hpp>
+
+// System include(s)
+#include <type_traits>
+
+namespace detray {
+
+/// @tparam T The type of the collection data, e.g. transforms
+/// @tparam context_t the context with which to retrieve the correct data.
+template <typename T, typename context_t>
+class single_store_base {
+    public:
+    using value_type = T;
+    using context_type = context_t;
+
+    /// Base vecmem types
+    // TO DISCUSS: change naming to builder_type?
+    using host_type = vecmem::vector<T>;
+    using buffer_type = vecmem::data::vector_buffer<T>;
+    using view_type = vecmem::data::vector_view<T>;
+    using device_type = vecmem::device_vector<T>;
+
+    using size_type = typename device_type::size_type;
+    using link_type = size_type;
+    using single_link = link_type;
+    using range_link = std::array<link_type, 2>;
+};
+
+template <typename T, typename context_t>
+DETRAY_HOST
+class single_store_host : single_store_base<T, context_t> {
+
+    public:
+    using _ = single_store_base<T, context_t>;
```

Comment:
Class inheritance didn't really work as I intended here, as I initially thought typedefs were inherited as well. Anyways, something similar to this still seems like it should be quite useful.

Reply:
You can inherit the typedefs in a dependent context by writing:

```cpp
using typename single_store_base<T, context_t>::size_type
...
```

Reply:
Thanks for the tip. I had tried that at one point but apparently messed it up somehow.